### ISSUE TITLE
Gitness on Docker with Network Macvlan

### ISSUE THREAD

**Issue by Drophoff** [harness/harness#3378]
I tried to run gitness on a machine with docker and the network mode macvlan.

This works with Drone 2:
- Start the drone-server with: `docker run --name drone-server --net=macvlan_netname -t -d drone/drone`
- Start the drone-runner-docker with: `docker run --name node01 --net=macvlan_netname -t -d  -e DRONE_RUNNER_NETWORKS=macvlan_netname -v /var/run/docker.sock:/var/run/docker.sock drone/drone-runner-docker`

Right now the clone failed because the clone step can not resolve http://host.docker.internal:3000.

This is expected with the used network mode because macvlan allows container-to-container but not container-to-host communication.

Drone allows this kind of setup via `DRONE_RUNNER_NETWORKS`. Is there a similar option within gitness to support macvlan?

**Comment by johannesHarness** [comment]
Hey @Drophoff, thanks for bringing this up!
We didn't add macvlan support to gitness as of now, we'll be looking into it 🚀

**Comment by Drophoff** [comment]
Sad to hear that an already supported feature is not available anymore.

**Comment by johannesHarness** [comment]
@Drophoff it's still available in drone, along with all its other great features (code is on `drone` branch at this point in time).
We are in the works of bringing gitness up to speed towards parity with drone, so it's much appreciated you are reporting things that are missing as of now 👍

**Comment by Drophoff** [comment]
This is true, but as mentioned in the readme, drone will be merged into gitness and will most likely not be developed further in the long run.

So I hope that the macvlan functionality will also be available in gitness in the long run or drone will be maintained as a standalone product.

Thank you very much for the quick and very helpful response!

**Comment by bradrydzewski** [comment]
@vistaarjuneja isn't there an "extra hosts" setting that will solve this problem?

**Comment by Drophoff** [comment]
**Regarding the `add-host`  option:**
The macvlan network mode allows the communication from container-to-container, but not from container-to-host.

The option '--add-host' will only add an IP with a corresponding domain name to the gitness container host file. As a result of this the gitness container knows the IP. But the problem is not, that the container is not knowing the IP. 

I would assume that the gitness container makes a call to the host because it expects to be run with default network mode with an accessible host, but I didn't checked the source code.

I already added the '-add-host' option and the result with the failed clone step remains.

**Regarding the Missing Feature**
Thanks for the clarification. This commitment was not preserved in this form in the previous response. Furthermore, I am very happy about this statement and the claim that comes with it.
